app/src/lib/WalletMultiButton.svelte

```diff
@@ -0,0 +1,27 @@
+<script lang="ts">
+	import { walletStore } from '$utils/walletStore';
+	import WalletButton from './WalletButton.svelte';
+	import WalletModalButton from './WalletModalButton.svelte';
+
+	function showAddress(store) {
+		const base58 = store.publicKey?.toBase58();
+		if (!store.wallet || !base58) return null;
+		return base58.slice(0, 4) + '..' + base58.slice(-4);
+	}
+
+	$: address = walletStore && showAddress($walletStore);
```

Comment:
this could be: 

`$: address = showAddress($walletStore);` 

What we solved with EventEmmiter3, was to prevent so many boilerplate :)